runtime_config_parser.go

```diff
@@ -0,0 +1,110 @@
+package dotnetcoreaspnetruntime
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/gravityblast/go-jsmin"
+)
+
+type framework struct {
+	Name    string `json:"name"`
+	Version string `json:"version"`
+}
+
+type RuntimeConfigParser struct{}
+
+func NewRuntimeConfigParser() RuntimeConfigParser {
+	return RuntimeConfigParser{}
+}
+
+func (p RuntimeConfigParser) Parse(glob string) (string, error) {
+	files, err := filepath.Glob(glob)
+	if err != nil {
+		return "", fmt.Errorf("%w: %q", err, glob)
+	}
+
+	if len(files) > 1 {
+		return "", fmt.Errorf("multiple *.runtimeconfig.json files present: %v", files)
+	}
+
+	if len(files) == 0 {
+		return "", os.ErrNotExist
+	}
+
+	path := files[0]
+
+	var data struct {
+		RuntimeOptions struct {
+			Framework  framework   `json:"framework"`
+			Frameworks []framework `json:"frameworks"`
+		} `json:"runtimeOptions"`
+	}
+
+	file, err := os.Open(path)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	buffer := bytes.NewBuffer(nil)
+	err = jsmin.Min(file, buffer)
+	if err != nil {
+		return "", err
+	}
```

Comment:
What's the rationale for minifying the JSON before decoding it?

Reply:
This is to remove comments inside of runtimeconfig.json (which is a thing that .NET allows)

Reply:
🤦 oh, right. thanks for the reminder.